src/components/views/elements/SettingsFlag.tsx

```diff
@@ -49,6 +48,7 @@ export default class SettingsFlag extends React.Component<IProps, IState> {
 
         this.state = {
             value: this.settingValue,
+            disabled: this.settingDisabled,
```

Comment:
Personally I'd find a regular method here clearer:

```suggestion
            disabled: this.isSettingDisabled(),
```